src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -109,6 +109,34 @@ public static function register($mode = 0)
             }
 
             $trace = debug_backtrace(true);
+
+            // Silence deprecation warnings about private service accessed
+            // from the service container if done so from a Test class.
+            // As of Symfony 4.1, there is a new TestContainer that allows
+            // fetching of private services within tests, so we no longer
+            // need to warn about this behavior:
+            //
+            // https://symfony.com/blog/new-in-symfony-4-1-simpler-service-testing
+            //
+            // NOTE: the event at the top of the stack $trace (index 0) should
+            // always be the PhpUnitBridge's DeprecationErrorHandler; the
+            // second event (index 1) should be the trigger_error() event;
+            // the third event (index 2) should be the actual source of the
+            // triggered deprecation notice; and the fourth event (index 3)
+            // represents the action that called the deprecated code. In the
+            // scenario that we want to suppress, the 4th event be an object
+            // instance of PHPUnit_Framework_TestCase.
+            if (isset($trace[3]) && isset($trace[3]['object'])) {
```

Comment:
The first is set is not needed

Reply:
I didn't originally have it, because I've never ever sen a stack track (particularly one for a framework) less than 4 elements, but I added it "just to be safe".  I will remove it.

Reply:
You don't need it because `isset` both check that `$trace[3]` exists and then that `$trace[3]['object']` exists: https://3v4l.org/4XGmL